### PR TITLE
Introduce AuthenticationInApp state machine

### PR DESCRIPTION
**Pull request by bakulf**
There are a couple of things I would like to do:
1. complete the state machine with more error handling [DONE]
2. support account creation
3. move _everything_ in the AIP listener [DONE]

┆Issue is synchronized with this [Jira Bug](https://mozilla-hub.atlassian.net/browse/VPN-797)

### CODE REVIEW

**Comment by bakulf** [comment]
This PR contains much more than what the title says:
- it has front-end code
- it implements sign in/up and the email verification
- error handling

Basically, what is missing is:
- password forgotten
- all the frontend implementation
- testing :)

File: src/authenticationinapp/authenticationinapplistener.cpp [hunk 1]
```diff
@@ -0,0 +1,592 @@
...
+                    });
+          });
+}
+
+void AuthenticationInAppListener::processErrorCode(int errorCode) {
+  AuthenticationInApp* aip = AuthenticationInApp::instance();
+  Q_ASSERT(aip);
+
+  switch (errorCode) {
+    case 101:  // Account already exists
```
Comment (oskirby) [inline src/authenticationinapp/authenticationinapplistener.cpp:359]: I dislike seeing naked integers hanging around in the code, I feel like these should be an `enum`.

**Review by oskirby — APPROVED** [review]
I think this is a great step forward, though I would suggest that some of the less common authentication states (such as password recovery and account creation) should get redirected to a browser session rather than trying to handle them in-app.

File: src/authenticationinapp/authenticationinapplistener.cpp (hunk 1 above)
Comment (bakulf) [reply src/authenticationinapp/authenticationinapplistener.cpp:358]: I agree. Would be OK-ish if I write a URL of where the numbers are defined?

Comment (oskirby) [reply src/authenticationinapp/authenticationinapplistener.cpp:358]: That would certainly be better

**Pull request merged**